### PR TITLE
Add a github actions workflow for PR

### PR DESCRIPTION
**Pull request by bernt-matthias**
So for each PR the workflow will check python linting and pytests is successful.

### CODE REVIEW

**Comment by bernt-matthias** [comment]
test fals currently because requirements.txt will be added only in https://github.com/yigbt/deepFPlearn/pull/11

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Test PR
```
Comment (halirutan) [inline .github/workflows/pr.yml:1]: I guess that's not the name :)

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: install pytest
+      run: pip install pytest
+    - name: install package
+      run: pip install -r requirements.txt .
+    - uses: ricardochaves/python-lint@v1.4.0
+      with:
+        python-root-list: "dfpl tests"
```
Comment (halirutan) [inline .github/workflows/pr.yml:17]: Another thing we need to do: See if all tests are up to date and pass. I can already see that they won't.

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: install pytest
+      run: pip install pytest
+    - name: install package
+      run: pip install -r requirements.txt .
+    - name: run tests
+      run: pytest tests
```
Comment (halirutan) [inline .github/workflows/pr.yml:27]: Missing linebreak

File: .github/workflows/pr.yml (hunk 1 above)
Comment (bernt-matthias) [reply .github/workflows/pr.yml:1]: Would `Test Pull Request` be better?

File: .github/workflows/pr.yml (hunk 2 above)
Comment (bernt-matthias) [reply .github/workflows/pr.yml:17]: Sure. Do you have a command line for me that starts the tests?

File: .github/workflows/pr.yml (hunk 3 above)
Comment (bernt-matthias) [reply .github/workflows/pr.yml:27]: Where? :)

Comment (bernt-matthias) [inline .github/workflows/pr.yml:27]:
```suggestion
    - name: run tests
      run: pytest tests 

```

Comment (halirutan) [reply .github/workflows/pr.yml]: After a year, I have no clue what I meant back then :)

File: .github/workflows/pr.yml (hunk 2 above)
Comment (halirutan) [reply .github/workflows/pr.yml]: It seems you possess the right street-smarts to have found this out already :)

File: .github/workflows/pr.yml (hunk 1 above)
Comment (halirutan) [reply .github/workflows/pr.yml]: Actually, I think I like it now.

**Comment by halirutan** [comment]
Looks good to me. There is one thing: Checking the order of imports with `isort` lets the action fail. I have no strong preference, but usually I import system libs first and references into our own package after that. If you also tend towards this, then you need to get rid of the `isort` checking. Otherwise, the imports where the "check sort action" breaks need to be fixed.

I think it's ready to be merged after that. Sorry, for taking so long. "Someone" had to remind me.

**Comment by bernt-matthias** [comment]
Now all works, except for typing :( Seems not trivial to fix. Should we:

1) skip for now and open a separate PR what adds and fixes typing
2) fix it here?

**Comment by halirutan** [comment]
Maybe this is a stupid question, but why is `mypy` not in the `requirements.txt` if you want to use it? As far as I can see from the GH Actions log, this here

image

indicates that it's just not installed. Am I missing something?

**Comment by bernt-matthias** [comment]
> Maybe this is a stupid question

There is no such thing as a stupid question :)

**Comment by halirutan** [comment]
The `requirements.txt` doesn't seem to work yet and I had to try it locally with the Python version GH action uses. The package `data-science-stubs` is nowhere to be found and replaced it with `data-science-types`. Also `flake8` is not available in version 6.0.0 for this python version. What worked is

```
black==23.3.0
data-science-types
flake8
isort==5.12.0
mypy==1.2.0
pip==22.0.4
pytest==7.1.1
types_setuptools
types-tensorflow
typing-extensions

-e .
```

**However**, running `mypy` results in 93 errors. So unless @mai00fti is willing to tidy up the code and make necessary changes, I recommend skipping the `mypy` test.

File: setup.py [hunk 4]
```diff
@@ -8,27 +8,25 @@
...
         "jsonpickle~=2.1",
         "matplotlib==3.5.1",
         "numpy==1.22.0",
         "pandas==1.4.2",
         "rdkit-pypi==2022.03.1",
         "scikit-learn==1.0.2",
-        "keras==2.6.0",
+        "keras==2.9.0",
         "tensorflow-gpu==2.9.3",
         "wandb~=0.12",
```
Comment (mai00fti) [inline setup.py:29]:
```suggestion
        "jsonpickle~=2.1.0",
        "matplotlib==3.5.1",
        "numpy==1.22.0",
        "pandas==1.4.2",
        "rdkit-pypi==2022.03.1",
        "scikit-learn==1.0.2",
        "keras==2.9.0",
        "tensorflow-gpu==2.9.3",
        "wandb~=0.12.0",
```

File: setup.py [hunk 5]
```diff
@@ -8,27 +8,25 @@
...
     url="https://github.com/yigbt/deepFPlearn",
     classifiers=[
         "Programming Language :: Python :: 3",
-        "Operating System :: OS Independent"
+        "Operating System :: OS Independent",
     ],
-    python_requires='>=3.6',
+    python_requires=">=3.6",
     install_requires=[
         "jsonpickle~=2.1",
```
Comment (mai00fti) [inline setup.py:21]:
```suggestion
        "jsonpickle~=2.1.0",
```

File: setup.py (hunk 4 above)
Comment (mai00fti) [inline setup.py:29]:
```suggestion
        "wandb~=0.12.0",
```

**Comment by bernt-matthias** [comment]
Hi @mai00fti can you check the [CLI tests](https://github.com/yigbt/deepFPlearn/actions/runs/4823386495/jobs/8591811953?pr=12#step:9:33). Learning seems to be to fast (compared to the runtime expectations that you mentioned), ie all three calls need 2:30min.

For `prediction` I see two warnings: 

```
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
WARNING  No training configuration found in save file, so the model was *not* compiled. Compile it manually.
```

Can they be ignored?

Also convert tells me that there is nothing to convert...

```
INFO     Convert all data files in /home/runner/work/deepFPlearn/deepFPlearn/example/data
INFO     Found 0 files to convert
```

File: setup.py [hunk 6]
```diff
@@ -8,27 +8,25 @@
...
     long_description_content_type="text/markdown",
     url="https://github.com/yigbt/deepFPlearn",
     classifiers=[
         "Programming Language :: Python :: 3",
-        "Operating System :: OS Independent"
+        "Operating System :: OS Independent",
     ],
-    python_requires='>=3.6',
+    python_requires=">=3.6",
     install_requires=[
```
Comment (mai00fti) [inline setup.py:20]:
```suggestion
    # all packages need for the final usage
    # for additional packages during development, use requirements.txt
    install_requires=[
```

File: requirements.txt [hunk 7]
```diff
@@ -1,14 +1,8 @@
-numpy~=1.19.1
-pandas~=1.4.2
-matplotlib~=3.5.1
-wandb~=0.12.11
-scikit-learn~=0.23.1
-jsonpickle~=1.4.1
-dataclasses~=0.8
-rdkit~=2020.03.4
+black~=23.3.0
```
Comment (mai00fti) [inline requirements.txt:1]:
```suggestion
# additional packages during development
# for all packages needed for the final usage use setup.py
black~=23.3.0
```

**Pull request merged**